### PR TITLE
Reworked BakedAnimation and Frames to not load for every entity

### PR DESCRIPTION
**Pull request by opl-**
By @Fadest 

Use data loaded from animation files on disk on load by cloning it when spawning new baked entities. Old behavior was to reload and deserialize the data on baked entity spawn.

### CODE REVIEW

File: src/main/java/com/mcmiddleearth/entities/entities/composite/animation/BakedAnimationTree.java [hunk 1]
```diff
@@ -104,15 +135,22 @@ public BakedAnimation getAnimation(String[] path) {
 
     public BakedAnimation getAnimation(BakedAnimationEntity entity) {
         String[] path = new String[] {
-                entity.getMovementType().name().toLowerCase(),
-                entity.getMovementSpeedAnimation().name().toLowerCase(),
-                ActionType.IDLE.name().toLowerCase()
-                //entity.getActionType().name().toLowerCase()
+            entity.getMovementType().name().toLowerCase(),
+            entity.getMovementSpeedAnimation().name().toLowerCase(),
+            getActionType(entity).name().toLowerCase(Locale.ROOT)
```
Comment (opl-) [inline src/main/java/com/mcmiddleearth/entities/entities/composite/animation/BakedAnimationTree.java]: I think this was done on purpose and will break animation lookups if changed?

**Review by opl- — COMMENTED** [review]
Seems there's no way to reload the animation data after it was loaded when the server was started. That's something we might need in the future.

File: src/main/java/com/mcmiddleearth/entities/entities/composite/animation/BakedAnimationTree.java (hunk 1 above)
Comment (Fadest) [reply src/main/java/com/mcmiddleearth/entities/entities/composite/animation/BakedAnimationTree.java]: port/magma-1.16 has the same getActionType() method, that's the one that should be used, the feature/rework_animation_frames branch also has one but without the goals implemented in magma

**Comment by opl-** [comment]
The merge conflict is caused by changes from https://github.com/CarperCreative/MCME-Entities/tree/fix/random-animations

**Comment by opl-** [comment]
Cleaned up and rebased onto `development` due to conflicts with `fix/random-animations`. Original in `feature/rework_animations_frames-original`. Cherry picked into `port/magma-1.16` in b42cb0896037363a705845c5bb89b266ef514eb4.

**Pull request closed without merging**